core/executor.py

```diff
@@ -595,41 +596,59 @@ def RunSubshell(self, node):
         return p.RunProcess(self.waiter, trace.ForkWait)
 
     def CaptureStdout(self, node):
-        # type: (command_t) -> Tuple[int, str]
+        # type: (command_t) -> Tuple[int, str, str]
 
         p = self._MakeProcess(node, self.exec_opts.inherit_errexit(),
                               self.exec_opts.errtrace())
         # Shell quirk: Command subs remain part of the shell's process group, so we
         # don't use p.AddStateChange(process.SetPgid(...))
 
         r, w = posix.pipe()
+        r2, w2 = posix.pipe()
         p.AddStateChange(process.StdoutToPipe(r, w))
+        p.AddStateChange(process.StderrToPipe(r2, w2))
 
         p.StartProcess(trace.CommandSub)
         #log('Command sub started %d', pid)
 
         chunks = []  # type: List[str]
+        chunks2 = []  # type: List[str]
         posix.close(w)  # not going to write
+        posix.close(w2)  # not going to write
+        eofcount = 0
         while True:
-            n, err_num = pyos.Read(r, 4096, chunks)
+            fds, w, exc = select.select([r,r2], [], [])
```

Comment:
@andychu I get the following error and don't know why 😬 

```
core/executor.py:621: error: Incompatible types in assignment (expression has type "List[Any]", variable has type "int")
```

Do you have an idea why this doesn't work here? I don't really get where it gets an `int` when I pass 3 lists...

(And don't worry, I'll make it a deparate function from `captureStdout`. I just wanted to test if it works that way on the phone and wanna clean it up on my laptop once it works)

Reply:
Oh sorry I didn't respond to this earlier ...

I think MyPy has problems inferring the type of the empty list `[]`.  So it just makes it  `List[Any]`

Sometimes I have to do this:

```
tmp = []  # type: List[int]
tmp2 = []  # type: List[int]
```